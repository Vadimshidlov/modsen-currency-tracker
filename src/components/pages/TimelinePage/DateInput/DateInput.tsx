import React, { Component } from "react";
import { DateInputType } from "@/types/TimeLinePageTypes/types";

class DateInput extends Component<DateInputType> {
    render() {
        const { className, ...rest } = this.props;

        return <input className={className} {...rest} type="date" data-testid="chart-date-input" />;
    }
}

export default DateInput;
