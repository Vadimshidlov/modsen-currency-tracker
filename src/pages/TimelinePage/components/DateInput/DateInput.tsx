import React, { Component } from "react";
import { DateInputType } from "@/types/TimeLinePageTypes";

class DateInput extends Component<DateInputType> {
    render() {
        const { className, ...rest } = this.props;

        return <input className={className} {...rest} type="date" data-testid="chart-date-input" />;
    }
}

export default DateInput;
