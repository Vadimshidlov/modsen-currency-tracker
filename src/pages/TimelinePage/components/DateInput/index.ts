export { default } from "@/pages/TimelinePage/components/DateInput/DateInput";
