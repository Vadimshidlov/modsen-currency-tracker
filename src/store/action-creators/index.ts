export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
