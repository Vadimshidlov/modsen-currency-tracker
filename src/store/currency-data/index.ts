export { mockMapData } from "@/store/currency-data/mockMapData";
export { mockBanksApiData } from "@/store/currency-data/mockBanksApiData";
export { mockStockData } from "@/store/currency-data/mockStockData";
