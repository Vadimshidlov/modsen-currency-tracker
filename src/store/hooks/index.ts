export { useActions } from "@/store/hooks/useActions";
export { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";
