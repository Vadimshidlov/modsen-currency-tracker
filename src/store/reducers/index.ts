import { combineReducers } from "redux";
import themeReducer from "@/store/reducers/themeReducer";
import latestCurrencyReducer from "@/store/reducers/latestCurrencyReducer";
import modalWindowReducer from "@/store/reducers/modalWindowReducer";
import mapDataReducer from "@/store/reducers/mapDataReducer";
import userLocationReducer from "@/store/reducers/locationAuthReducer";

export const rootReducer = combineReducers({
    theme: themeReducer,
    currency: latestCurrencyReducer,
    modalWindow: modalWindowReducer,
    mapData: mapDataReducer,
    userLocation: userLocationReducer,
});

export type RootStateType = ReturnType<typeof rootReducer>;
