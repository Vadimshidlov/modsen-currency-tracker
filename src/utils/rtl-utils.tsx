import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
