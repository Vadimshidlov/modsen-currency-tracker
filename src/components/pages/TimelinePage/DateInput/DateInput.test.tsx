import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { fireEvent } from "@testing-library/react";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import DateInput from "@/components/pages/TimelinePage/DateInput/DateInput";

describe("CurrencyStocks component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<DateInput className="date-input" />);

        const inputElement = screen.getByTestId("chart-date-input");

        expect(inputElement).toBeInTheDocument();
        expect(inputElement).toHaveClass("date-input");
        expect(inputElement).toHaveAttribute("type", "date");
    });

    it("fires onChange event with the correct value", () => {
        const onChange = jest.fn();

        renderWithProviders(<DateInput className="date-input" onChange={onChange} />);

        const inputElement = screen.getByTestId("chart-date-input");
        const testDate = "2024-03-06";
        fireEvent.change(inputElement, { target: { value: testDate } });

        expect(onChange).toHaveBeenCalled();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(<DateInput className="date-input" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <DateInput className="date-input" />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
