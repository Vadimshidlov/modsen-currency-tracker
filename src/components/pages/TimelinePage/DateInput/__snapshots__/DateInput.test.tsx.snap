// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyStocks component tests should match snapshot 1`] = `
<input
  className="date-input"
  data-testid="chart-date-input"
  type="date"
/>
`;

exports[`CurrencyStocks component tests snapshot matches 1`] = `
<DocumentFragment>
  <input
    class="date-input"
    data-testid="chart-date-input"
    type="date"
  />
</DocumentFragment>
`;
